# Data Science Jobs Scraper

In [19]:
import re
import requests
import numpy as np
import pandas as pd
from bs4 import BeautifulSoup

### Import Target Keywords

In [20]:
df = pd.read_csv("wd_skills.csv")

df["Skill"] = df["Skill"].str.lower()
df["Category"] = df["Category"].str.lower()
df['total_jobs'] = 0

print("Total Skills: ", df.shape[0], " technologies")
df.head()

Total Skills:  60  technologies


Skill     Category  total_jobs
0  javascript  programming           0
1      python  programming           0
2        java  programming           0
3       scala  programming           0
4        rust  programming           0

In [21]:
skills = df['Skill'].to_list()

print(skills)

['javascript', 'python', 'java', 'scala', 'rust', 'c', 'go', 'php', 'ruby', 'swift', 'bash', 'html', 'css', 'sass', 'markdown', 'aws', 'azure', 'google cloud', 'react', 'angular', 'vue', 'heroku', 'netlify', 'github pages', 'firebase', 'git', 'github', 'bitbucket', 'gitlab', 'wordpress', 'strapi', 'contentful', 'figma', 'adobe xd', 'sql', 'nosql', 'mongo', 'mysql', 'postgresql', 'maria', 'oracle 12c', 'microsft sql', 'db2', 'graphql', 'sockets', 'digital ocean', 'cpanel', 'docker', 'kubernetes', 'nginx', 'apache', 'rest', 'django', 'ruby on rails', 'node', 'laravel', 'flask', 'cloudflare', 'nextjs', 'gatsby']


## Indeed

In [22]:
url = f'https://www.indeed.com/jobs?q=%22web+development%22&l=United+States'

try:
    r = requests.get(url)
    soup = BeautifulSoup(r.text, 'html.parser')
    count_str = soup.find('div', id="searchCountPages").get_text()
    numb = count_str.split()
    indeed_total = int(re.sub(",", "", numb[-2]))

except Exception as e:
    print(f'error: {e}')

print(indeed_total, "Web Development Jobs")

9825 Web Development Jobs


In [23]:
indeed_list = []

for skill in skills:
    url = f'https://www.indeed.com/jobs?q=%22web+development%22+%22{skill}%22&l=United+States'
    
    try:
        r = requests.get(url)
        soup = BeautifulSoup(r.text, 'html.parser')
        count_str = soup.find('div', id="searchCountPages").get_text()
        numb = count_str.split()
        indeed_list.append(numb[-2])
    except Exception as e:
        indeed_list.append(0)
        print(f'error: {e}')
        print(skill)

print(indeed_list)

error: 'NoneType' object has no attribute 'get_text'
microsft sql
['5,141', '1,683', '2,359', '93', '31', '778', '733', '1,320', '617', '157', '197', '3,760', '3,787', '477', '5', '1,404', '822', '132', '2,210', '1,892', '601', '51', '1', '2', '21', '1,770', '430', '213', '162', '920', '2', '15', '89', '85', '2,808', '441', '95', '1,019', '386', '8', '16', 0, '70', '170', '20', '4', '20', '710', '477', '108', '379', '1,432', '274', '242', '966', '140', '114', '10', '12', '29']


In [24]:
indeed = []

for i in indeed_list:
    indeed.append(int(re.sub(",", "", str(i))))

df['total_jobs'] = df['total_jobs'] + indeed_total
df['indeed'] = indeed
df.head()

Skill     Category  total_jobs  indeed
0  javascript  programming        9825    5141
1      python  programming        9825    1683
2        java  programming        9825    2359
3       scala  programming        9825      93
4        rust  programming        9825      31

## Simply Hired

In [25]:
url = f'https://www.simplyhired.com/search?q=%22web+development%22&l=United+States'
    
try:
    r = requests.get(url)
    soup = BeautifulSoup(r.text, 'html.parser')
    count_str = soup.find('span', class_="CategoryPath-total").get_text()
    simply_total = int(re.sub(",", "", count_str))

except Exception as e:
    print(f'error: {e}')

print(simply_total, "Web Development Jobs")

7728 Web Development Jobs


In [26]:
simply_list = []

for skill in skills:
    url = f'https://www.simplyhired.com/search?q=%22web+development%22+%22{skill}%22&l=United+States'
    
    try:
        r = requests.get(url)
        soup = BeautifulSoup(r.text, 'html.parser')
        count_str = soup.find('span', class_="CategoryPath-total").get_text()
        simply_list.append(count_str)
    except Exception as e:
        simply_list.append(0)
        print(f'error: {e}')
        print(skill)
        
print(simply_list)

error: 'NoneType' object has no attribute 'get_text'
microsft sql
['4,001', '1,350', '1,883', '73', '26', '597', '633', '1,153', '497', '100', '155', '3,024', '3,017', '391', '5', '1,140', '673', '122', '1,633', '1,380', '500', '44', '1', '2', '19', '1,477', '375', '164', '140', '830', '2', '13', '77', '59', '2,320', '348', '74', '853', '319', '5', '11', 0, '57', '155', '17', '4', '14', '558', '355', '94', '331', '1,134', '211', '196', '795', '120', '85', '9', '11', '26']


In [28]:
simply = []

for i in simply_list:
    simply.append(int(re.sub(",", "", str(i))))

df['total_jobs'] = df['total_jobs'] + simply_total
df['simply'] = simply
df.head()

Skill     Category  total_jobs  indeed  simply
0  javascript  programming       17553    5141    4001
1      python  programming       17553    1683    1350
2        java  programming       17553    2359    1883
3       scala  programming       17553      93      73
4        rust  programming       17553      31      26

## Glassdoor

In [29]:
url = f'https://www.glassdoor.com/Job/jobs.htm?sc.keyword="web+development"+"%22python%22"'

r = requests.get(url)
print(r.status_code)

403


**Status code is 403 (Forbidden)**. Therefore, we can not scrape content from Glassdoor. Optionally, we can try using **Selenium** in the future.

## Zip Recruiter

In [30]:
url = f'https://www.ziprecruiter.com/candidate/search?search=web+development&location=United+States'

r = requests.get(url)
print(r.status_code)

403


**Status code is 403 (Forbidden)**. Therefore, we can not scrape content from Glassdoor. Optionally, we can try using **Selenium** in the future.

## Career Builder

In [31]:
url = 'https://www.careerbuilder.com/jobs?keywords=Web+Development&location=United+States'

try:
    r = requests.get(url)
    soup = BeautifulSoup(r.text, 'html.parser')
    count_str = soup.find('h1', class_="fz1rem").get_text()
    career_total = int(re.sub(",", "", count_str.split()[0]))

except Exception as e:
    print(f'error: {e}')

print(career_total, "Web Development Jobs")

15858 Web Development Jobs


In [32]:
career_list = []

for skill in skills:
    keyword = str(re.sub(",", "+", skill))
    url = f'https://www.careerbuilder.com/jobs?keywords={keyword}+Web+Development&location=United+States'
    
    try:
        r = requests.get(url)
        soup = BeautifulSoup(r.text, 'html.parser')
        count_str = soup.find('h1', class_="fz1rem").get_text()
        career_list.append(int(re.sub(",", "", count_str.split()[0])))
    
    except Exception as e:
        career_list.append(0)
        print(f'error: {e}')
        print(skill)
        
print(career_list)

error: 'NoneType' object has no attribute 'get_text'
strapi
error: 'NoneType' object has no attribute 'get_text'
microsft sql
error: 'NoneType' object has no attribute 'get_text'
gatsby
[4787, 2233, 5192, 195, 13, 3282, 1206, 535, 504, 147, 387, 4016, 3443, 344, 11, 2638, 1966, 450, 1879, 2257, 390, 39, 4, 14, 28, 2351, 538, 305, 191, 217, 0, 7, 54, 45, 5196, 926, 133, 832, 478, 12, 19, 0, 287, 178, 65, 2, 4, 1111, 707, 121, 810, 3084, 127, 164, 1183, 50, 96, 3, 11, 0]


In [33]:
career = []

for i in career_list:
    career.append(int(i))

df['total_jobs'] = df['total_jobs'] + career_total
df['career'] = career
df.head()

Skill     Category  total_jobs  indeed  simply  career
0  javascript  programming       33411    5141    4001    4787
1      python  programming       33411    1683    1350    2233
2        java  programming       33411    2359    1883    5192
3       scala  programming       33411      93      73     195
4        rust  programming       33411      31      26      13

### Export Dataset

In [34]:
print(df.shape)
df.to_csv("wd_scraped_data.csv")

(60, 6)
